kernel/thread.c

```diff
@@ -381,7 +440,11 @@ void z_setup_new_thread(struct k_thread *new_thread,
 	k_spin_unlock(&lock, key);
 #endif
 #ifdef CONFIG_THREAD_NAME
-	new_thread->name = name;
+	if (name) {
```

Comment:
IIRC MISRA rules disallow implicit cast to bool and require "name != NULL" here.